### PR TITLE
test(slider): Add Jasmine foundation tests.

### PR DESCRIPTION
**Pull request by joyzhong**
Project import generated by Copybara.

### CODE REVIEW

**Comment by joyzhong** [comment]
Please hold off on reviewing until I get the slider import submitted in g3.

File: packages/mdc-dialog/package.json [hunk 1]
```diff
@@ -21,7 +21,6 @@
   "dependencies": {
     "@material/animation": "^4.0.0",
     "@material/base": "^4.0.0",
-    "@material/button": "^4.0.0",
```
Comment (abhiomkar) [inline packages/mdc-dialog/package.json]: This was fixed on latest master branch. Rebase with master.

File: packages/mdc-slider/foundation.ts [hunk 2]
```diff
@@ -27,13 +27,13 @@ import {EventType, SpecificEventListener} from '@material/base/types';
 import {MDCSliderAdapter} from './adapter';
 import {cssClasses, numbers, strings} from './constants';
 
-type UpEventType = 'mouseup' | 'pointerup' | 'touchend';
-type DownEventType = 'mousedown' | 'pointerdown' | 'touchstart';
-type MoveEventType = 'mousemove' | 'pointermove' | 'touchmove';
-type MouseLikeEvent = MouseEvent | PointerEvent | TouchEvent;
+type UpEventType = 'mouseup'|'pointerup'|'touchend';
```
Comment (abhiomkar) [inline packages/mdc-slider/foundation.ts]: Can we revert changes to the component?

Comment (joyzhong) [reply packages/mdc-slider/foundation.ts]: Keeping as is, since these are style fixes from internal `g4 fix`.

**Pull request merged**